### PR TITLE
feat: Use Api Grpc Channel Factory

### PR DESCRIPTION
**Pull request by dbrasseur-aneo**
Use the Api implementation of Grpc channel creation

### CODE REVIEW

File: Client/src/Common/Submitter/ClientServiceConnector.cs
Comment (aneojgurhem) [inline Client/src/Common/Submitter/ClientServiceConnector.cs:91]: clientP12 missing in doc

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/ArmoniKPayloadSerialization.cs
Comment (aneojgurhem) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/ArmoniKPayloadSerialization.cs:40]: why is this changing ?

Comment (aneojgurhem) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/ArmoniKPayloadSerialization.cs:40]: that's because in the root build.props, langage is latest so 11 today. You need to install dotnet 7

Comment (dbrasseur-aneo) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/ArmoniKPayloadSerialization.cs:40]: Since we expect to build in dotnet 6 (as it's the latest target in the csproj), shouldn't we keep it as dotnet 6 compatible ?

Comment (aneojgurhem) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/ArmoniKPayloadSerialization.cs:40]: I think so, we should fix the langage version to 10 but for the whole project. Thus, not this pr

File: Client/src/Common/Submitter/ClientServiceConnector.cs
Comment (ddubuc) [inline Client/src/Common/Submitter/ClientServiceConnector.cs:55]: Could you run the clientwith Dotnet Frameworks 4.6. I think this one compile but fail at runtime on windows

**Review by ddubuc — DISMISSED** [review]
Please be sure it will work on windows 10 before the merge

File: Client/src/Common/Submitter/ClientServiceConnector.cs
Comment (dbrasseur-aneo) [reply Client/src/Common/Submitter/ClientServiceConnector.cs:55]: Indeed, even after the changes in Api, .net framework doesn't support http2, so back to the drawing board in API

**Comment by dbrasseur-aneo** [comment]
I will update Api to 3.8.1 when https://github.com/aneoconsulting/ArmoniK.Api/pull/227 is released

File: Client/src/Common/ArmoniK.DevelopmentKit.Client.Common.csproj
Comment (lemaitre-aneo) [inline Client/src/Common/ArmoniK.DevelopmentKit.Client.Common.csproj:11]: This should be changed to a stable release before merging.

**Review by lemaitre-aneo — DISMISSED** [review]

**Comment by dbrasseur-aneo** [comment]
Do not merge until we have automated tests

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/appsettings.json
Comment (ddubuc) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/appsettings.json:4]: revert change in this file

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/appsettings.json
Comment (ddubuc) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/appsettings.json:53]: Same here

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/ArmoniK.EndToEndTests.Common.csproj
Comment (ddubuc) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/ArmoniK.EndToEndTests.Common.csproj:16]: be sure you will add feat: in the PR of Ext.C# !!

**Review by ddubuc — CHANGES_REQUESTED** [review]

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/ArmoniK.EndToEndTests.Common.csproj
Comment (dbrasseur-aneo) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/ArmoniK.EndToEndTests.Common.csproj:16]: Already done

**Review by lemaitre-aneo — APPROVED** [review]

File: .github/workflows/build.yml
Comment (dbrasseur-aneo) [reply .github/workflows/build.yml:240]: I need to work on the ci anyway, will do that in another pr

**Pull request merged**
